composer/callbacks/checkpoint_saver.py

```diff
@@ -520,11 +549,18 @@ def _save_checkpoint(self, state: State, logger: Logger):
                     this_rank_saves_symlinks = dist.get_global_rank() == 0 or not state.fsdp_sharded_state_dict_enabled
                     if this_rank_saves_symlinks:
                         create_symlink_file(src_path, symlink_filename)
-                        logger.upload_file(
-                            remote_file_name=symlink_name,
-                            file_path=symlink_filename,
-                            overwrite=True,
-                        )
+                        if self.use_remote_uploader_v2 and self.remote_uploader is not None:
+                            self.remote_uploader.upload_file_async(
+                                remote_file_name=symlink_name,
+                                file_path=symlink_filename,
+                                overwrite=True,
```

Comment:
ok this makes sense for a PR, where we just simplify the remoteuploader and don't change any functionality, but eventually we want to have the remoteuploader manage symlinks. That way we can actually avoid race conditions